packages/server/modules/workspaces/services/regions.ts

```diff
@@ -0,0 +1,39 @@
+import { GetRegions } from '@/modules/multiregion/domain/operations'
+import {
+  AssignRegion,
+  GetAvailableRegions,
+  UpsertRegionAssignment,
+  DeleteAllRegionAssignments
+} from '@/modules/workspaces/domain/operations'
+import { WorkspaceRegionAssignmentError } from '@/modules/workspaces/errors/regions'
+
+export const getAvailableRegionsFactory =
+  (deps: { getRegions: GetRegions }): GetAvailableRegions =>
+  async () => {
+    // TODO: Gatekeeper checks here? For now just returning all server regions
+    return await deps.getRegions()
+  }
+
+export const assignRegionFactory =
+  (deps: {
+    getAvailableRegions: GetAvailableRegions
+    upsertRegionAssignment: UpsertRegionAssignment
+    deleteAllRegionAssignments: DeleteAllRegionAssignments
+  }): AssignRegion =>
+  async (params) => {
+    const { workspaceId, regionKey } = params
+
+    const availableRegions = await deps.getAvailableRegions({ workspaceId })
+    if (!availableRegions.find((region) => region.key === regionKey)) {
+      throw new WorkspaceRegionAssignmentError(
+        'Specified region not available for workspace',
+        {
+          info: { params }
+        }
+      )
+    }
+
+    // For now, only 1 region - delete old assignments
+    await deps.deleteAllRegionAssignments({ workspaceId })
+    await deps.upsertRegionAssignment({ workspaceId, regionKey })
```

Comment:
In here, we need to copy the workspace record into the region DB, so that projects in the region DB, can use the `workspaceId` foreign key